components/org.wso2.carbon.identity.authz.valve/src/main/java/org/wso2/carbon/identity/authz/valve/AuthorizationValve.java

```diff
@@ -122,6 +122,13 @@ public void invoke(Request request, Response response) throws IOException, Servl
                 authorizationContext.addParameter(OAUTH2_VALIDATE_SCOPE, authenticationContext.getParameter(OAUTH2_VALIDATE_SCOPE));
                 authorizationContext.addParameter(VALIDATE_LEGACY_PERMISSIONS,
                         authenticationContext.getParameter(VALIDATE_LEGACY_PERMISSIONS));
+                Pattern patternTenantPerspective = Pattern.compile("^/t/[^/]+/o/[a-f0-9\\-]+?");
+                if (patternTenantPerspective.matcher(requestURI).find()) {
+                    int startIndex = requestURI.indexOf("/o/") + 3;
+                    int endIndex = requestURI.indexOf("/", startIndex);
+                    String resourceOrgId = requestURI.substring(startIndex, endIndex);
+                    authorizationContext.addParameter("resourceOrgId", resourceOrgId);
```

Comment:
Better to go for a constant for "resourceOrgId" since it is used in another place as well

Reply:
"resourceOrgId" is used in two seperate component where no common places are available to add this. Hence keeping this as it is.

Reply:
Isn't that one depend on another too? if there is a dependency add to the base component

Reply:
Updated